block/manager.go

```diff
@@ -443,6 +455,11 @@ func (m *Manager) BatchRetrieveLoop(ctx context.Context) {
 
 					// Update lastBatchHash only if the batch contains transactions
 					if batch.Transactions != nil {
+						err := m.store.SetMetadata(ctx, LastBatchHashKey, h)
```

Comment:
do you think we can avoid writing to disk on every update? i think we can keep it in-memory and only write to db upon shutdown? how about moving this to:
```
func (n *FullNode) OnStop()
```
and calling `n.blockManager.Stop()` and inserting this update logic in the `Stop()` method? 
trying to avoid expensive disk writes.

Reply:
opened an issue https://github.com/rollkit/rollkit/issues/1993